platform/mellanox/mlnx-platform-api/sonic_platform/psu.py

```diff
@@ -137,6 +161,26 @@ def _read_generic_file(self, filename, len):
         return result
 
 
+    def get_model(self):
+        """
+        Retrieves the model number (or part number) of the device
+
+        Returns:
+            string: Model/part number of devic
+        """
+        return self._read_vpd_file(self.psu_vpd)[PN_VPD_FIELD]
```

Comment:
I would suggest reading only one time in __init__(), and cache it with a variable, each time this function called just return the variable, no need to read the file every time.